pydantic/config.py

```diff
@@ -555,15 +555,49 @@ class Transaction(BaseModel):
     3. Using `'never'` we would have gotten `user=SubUser(hobbies=['scuba diving'], sins=['lying'])`.
     """
 
+    val_date_or_time_unit: Literal['seconds', 'milliseconds', 'infer']
+    """The unit to assume for validating date or time strings. Defaults to `'infer'`.
+    Only applies when the date / time like input is an int / float.
+
+    The "epoch" reference below refers to the Unix epoch, which is 1970-01-01 00:00:00 UTC.
+
+    - `'seconds'` will validate date or time strings as seconds since the epoch.
+    - `'milliseconds'` will validate date or time strings as milliseconds since the epoch.
+    - `'infer'` will infer the unit from the string based on unix time:
+        i.e. seconds (if >= -2e10 and <= 2e10) or milliseconds (if < -2e10or > 2e10) since the epoch.
+    """
+
+    # TODO: does infer even make sense here? I guess in the context of float?
```

Comment:
Yes I think we want it for symmetry, if the default is `infer` then things round-trip properly. Otherwise we end up changing the scale for users when they validate then dump.